### ISSUE TITLE
Predicate built-in functions: valid code rejected by lint, invalid code accepted

### ISSUE THREAD

**Issue by sandraros** [abaplint/abaplint#2171]
There was the following lint error while checking [abap2xlsx#805](https://github.com/sapmentors/abap2xlsx/pull/805):
```
Statement does not exist in ABAPv702(or a parser error), "IF".
```

Concerned code (using the predicate built-in function [`contains`](https://help.sap.com/doc/abapdocu_latest_index_htm/latest/en-US/index.htm?file=abencontains_functions.htm)):
```
      IF contains( val = <fs_sheet_string>-string_value start = ` ` )
            OR contains( val = <fs_sheet_string>-string_value end = ` ` ).
```
I think the syntax is valid in ABAP 7.02. At least it's accepted in 7.52.

If I force it to this invalid code (ABAP syntax error in 7.52), lint doesn't report an error:
```
      IF contains( val = <fs_sheet_string>-string_value start = ` ` ) = abap_true
            OR contains( val = <fs_sheet_string>-string_value end = ` ` ) = abap_true.
```

Workaround which both works in ABAP (at least 7.52) and in lint (ABAPv702):
```
      IF boolc( contains( val = <fs_sheet_string>-string_value start = ` ` ) ) = abap_true
            OR boolc( contains( val = <fs_sheet_string>-string_value end = ` ` ) ) = abap_true.
```

**Comment by larshp** [comment]
Predicative method calls/expressions was introduced in 740sp08,

https://help.sap.com/doc/abapdocu_752_index_htm/7.52/en-US/abennews-740_sp08-expressions.htm#!ABAP_MODIFICATION_1@1@

**Comment by sandraros** [comment]
The **built-in** predicate functions were introduced in 7.02:
https://help.sap.com/doc/abapdocu_752_index_htm/7.52/en-US/index.htm?file=abennews-71-logical_expressions.htm

Maybe lint could tolerate these built-in predicate functions, considering that there are very few ones (`matches`, `contains`, `contains_any_of`, `contains_any_not_of`, `line_exists`), low risk to be confused with functional predicative methods?

**Comment by larshp** [comment]
hmm, yea

would like to test it first, but I dont have access to any 702 system

**Comment by larshp** [comment]
https://github.com/abaplint/abaplint/pull/2560/commits/b0e8bc5ad59e38dacff7b7ea61362602ac5a4957 adds information on what is builtin predicate functions
